NEMO/interlocks.py

```diff
@@ -201,6 +201,44 @@ def _send_command(self, interlock: Interlock_model, command_type: Interlock_mode
 			sock.close()
 
 
+class ProXrInterlock(Interlock):
+	"""Very basic support for ProXR relay controllers.
+	Supports on/off for all relays in bank 1.
+	See https://ncd.io/proxr-quick-start-guide/ for more about ProXR.
+	"""
+	# supported proxr commands
+	PXR_CMD_B1_OFF = (254, 129, 1)
+	PXR_CMD_B1_ON = (254, 130, 1)
+
+	def _send_bytes(self, relay, proxrcmd):
+		"""Returns the response from the relay controller.
+		Argument relay is a connected socket object.
+		Argument proxrcmd (the ProXR command) is an iterable of 8-bit integers.
+		"""
+		# make sure that all bytes get sent
+		bytes_sent = 0
+		while bytes_sent < len(proxrcmd):
+			bytes_sent += relay.send(bytes(proxrcmd[bytes_sent:]))
+		# relay responses can include erroneous data
+		# only the last byte of the response is important
+		return relay.recv(64)[-1]
+
+	def _send_command(self, interlock: Interlock_model, command_type: Interlock_model.State) -> Interlock_model.State:
+		"""Returns and sets NEMO locked/unlocked state."""
+		state = Interlock_model.State.UNKNOWN
+		try:
+			with socket.create_connection((interlock.card.server, interlock.card.port), 10) as relay:
```

Comment:
I prefer relay_socket here as well

Reply:
I created relay_socket as a class attribute, it cleaned up the code overall. When you turn these relays on/off the response is always 85. It isn't a terribly useful response. But I am following up and asking for the relay state so that the NEMO state should better reflect reality.